src/SqlPersistence/ScriptLocation.cs

```diff
@@ -20,20 +19,15 @@ static string GetCurrentDirectory(ReadOnlySettings settings)
         {
             return scriptDirectory;
         }
-        var entryAssembly = Assembly.GetEntryAssembly();
-        if (entryAssembly == null)
+
+        var baseDir = AppDomain.CurrentDomain.BaseDirectory;
+        var scriptDir = Path.Combine(baseDir, ScriptFolder);
+        //if the app domain base dir contains the scripts folder, return it. Otherwise add "bin" to the base dir so that web apps work correctly
+        if (Directory.Exists(scriptDir))
         {
-            var baseDir = AppDomain.CurrentDomain.BaseDirectory;
-            var scriptDir = Path.Combine(baseDir, ScriptFolder);
-            //if the app domain base dir contains the scripts folder, return it. Otherwise add "bin" to the base dir so that web apps work correctly
-            if (Directory.Exists(scriptDir))
-            {
-                return baseDir;
-            }
-            return Path.Combine(baseDir, "bin");
+            return baseDir;
         }
-        var codeBase = entryAssembly.CodeBase;
-        return Directory.GetParent(new Uri(codeBase).LocalPath).FullName;
```

Comment:
I'd almost consider suggesting switching `GetEntryAssembly` to `GetExecutingAssembly` and leaving the rest of the code as-is, but this line also isn't quite correct. The `CodeBase` property might look like a uri, but it's not actually valid, so passing it to the `Uri` constructor can fail in some cases.

And you can't rely on `Location` instead because that will reflect any shadow copying that could be going on, which would be the wrong path.